editor/editor_properties_array_dict.cpp

```diff
@@ -301,108 +364,387 @@ void EditorPropertyArray::update_property() {
 
 				// atomic types
 				case Variant::BOOL: {
-
 					prop = memnew(EditorPropertyCheck);
 
 				} break;
 				case Variant::INT: {
-					EditorPropertyInteger *ed = memnew(EditorPropertyInteger);
```

Comment:
This is waay to hacky, too much code that is already in the plugins, if you are going to go full way, just reuse what is there